### PR TITLE
Feat/met 1750 link checking support for edm external

### PR DESCRIPTION
**Pull request by jochen-vermeulen**

### CODE REVIEW

File: metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/RdfBindingFactoryProvider.java [hunk 1]
```diff
@@ -7,16 +7,18 @@
 import org.jibx.runtime.JiBXException;
 
 /**
- * This is the super class of all RDF converters, providing an instance of {@link IBindingFactory}.
+ * This class maintains an instance of {@link IBindingFactory} which it can make available upon
+ * request.
  */
-class RdfConverter {
+class RdfBindingFactoryProvider {
```
Comment (stzanakis) [inline metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/RdfBindingFactoryProvider.java]: final?

File: metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/RdfBindingFactoryProvider.java [hunk 2]
```diff
@@ -7,16 +7,18 @@
...
 
   private static IBindingFactory rdfBindingFactory;
 
   /**
    * Constructor.
    */
-  RdfConverter() {}
+  RdfBindingFactoryProvider() {
```
Comment (stzanakis) [inline metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/RdfBindingFactoryProvider.java]: private?

File: metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/RdfDeserializerWithXPath.java [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    final Map<String, List<UrlType>> urls = new HashMap<>();
+    Function<String, List<UrlType>> listProd = k -> new ArrayList<>();
+    for (UrlType type : allowedUrlTypes) {
+      final Set<String> urlsForType = getUrls(document, type);
+      for (String url : urlsForType) {
+        urls.computeIfAbsent(url, listProd).add(type);
+      }
+    }
+
+    // Convert them to resource entries and return.
```
Comment (stzanakis) [inline metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/RdfDeserializerWithXPath.java]: Should this comment be here?

File: metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/RdfBindingFactoryProvider.java (hunk 1 above)
Comment (jochen-vermeulen) [reply metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/RdfBindingFactoryProvider.java]: True, this class now becomes a utils class with only static methods. I changed it.

File: metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/RdfBindingFactoryProvider.java (hunk 2 above)
Comment (jochen-vermeulen) [reply metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/RdfBindingFactoryProvider.java]: See above. Fixed.

File: metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/RdfDeserializerWithXPath.java (hunk 3 above)
Comment (jochen-vermeulen) [reply metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/RdfDeserializerWithXPath.java]: No, it shouldn't.

**Pull request merged**